sway-core/src/semantic_analysis/ast_node/declaration/impl_trait.rs

```diff
@@ -864,29 +986,112 @@ fn type_check_const_decl(
+                expected: engines.help_out(const_decl_signature_type_id).to_string(),
             }),
         );
     }
 
     Ok(const_decl)
 }
 
+fn type_check_type_decl(
+    handler: &Handler,
+    mut ctx: TypeCheckContext,
+    type_decl: &TraitTypeDeclaration,
+    trait_name: &CallPath,
+    is_contract: bool,
+    impld_type_ids: &ItemMap,
+    type_checklist: &BTreeMap<Ident, ty::TyTraitType>,
+) -> Result<ty::TyTraitType, ErrorEmitted> {
+    let engines = ctx.engines();
+    let type_engine = engines.te();
+    let self_type = ctx.self_type();
+
+    let mut ctx = ctx
+        .by_ref()
+        .with_help_text("")
+        .with_type_annotation(type_engine.insert(engines, TypeInfo::Unknown));
+
+    let interface_name = || -> InterfaceName {
+        if is_contract {
+            InterfaceName::Abi(trait_name.suffix.clone())
+        } else {
+            InterfaceName::Trait(trait_name.suffix.clone())
+        }
+    };
+
+    // type check the type declaration
+    let type_decl = ty::TyTraitType::type_check(handler, ctx.by_ref(), type_decl.clone())?;
+
+    let type_name = type_decl.name.clone();
+
+    // Ensure that there aren't multiple definitions of this type
+    if impld_type_ids.contains_key(&(type_name.clone(), self_type)) {
+        return Err(handler.emit_err(CompileError::MultipleDefinitionsOfType {
+            name: type_name.clone(),
+            span: type_name.span(),
+        }));
+    }
+
+    // Ensure that the type checklist contains this type.
+    let mut type_decl_signature = match type_checklist.get(&type_name) {
+        Some(type_decl) => type_decl.clone(),
+        None => {
+            return Err(
+                handler.emit_err(CompileError::TypeNotAPartOfInterfaceSurface {
+                    name: type_name.clone(),
+                    interface_name: interface_name(),
+                    span: type_name.span(),
+                }),
+            );
+        }
+    };
+
+    // replace instances of `TypeInfo::SelfType` with a fresh
+    // `TypeInfo::SelfType` to avoid replacing types in the stub constant
+    // declaration
+    type_decl_signature.replace_self_type(engines, self_type);
+
+    // unify the types from the constant with the constant signature
+    /*if !type_engine.get(const_decl.type_ascription.type_id).eq(
+        &type_engine.get(const_decl_signature.type_ascription.type_id),
+        engines,
+    ) {
+        errors.push(CompileError::MismatchedTypeInInterfaceSurface {
+            interface_name: interface_name(),
+            span: const_decl.span.clone(),
+            decl_type: "constant".to_string(),
+            given: engines
+                .help_out(const_decl.type_ascription.type_id)
+                .to_string(),
+            expected: engines
+                .help_out(const_decl_signature.type_ascription.type_id)
+                .to_string(),
+        });
+        return err(warnings, errors);
+    }*/
```

Comment:
Is this meant to be commented out?

Reply:
Removed the unnecessary code. This was copied from the const implementation but is not required for associated types.